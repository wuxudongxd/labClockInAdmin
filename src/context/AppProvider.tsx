import { PropsWithChildren } from "react";
import { QueryClient, QueryClientProvider } from "react-query";
import { BrowserRouter } from "react-router-dom";

/**
 * 使用react query进行服务端数据的缓存、更新、组件间共享
 */
// 提供 QueryClientProvider
const AppProvider = ({ children }: PropsWithChildren<{}>) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        refetchOnWindowFocus: false,
      },
    },
  });

  return (
    <QueryClientProvider client={queryClient}>
      <BrowserRouter>{children}</BrowserRouter>
    </QueryClientProvider>
  );
};

export default AppProvider;
