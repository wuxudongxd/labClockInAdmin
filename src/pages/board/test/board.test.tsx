import { render, screen } from "@testing-library/react";
import { describe, it, vi } from "vitest";
import AppProvider from "~/context/AppProvider";
import mountTest from "~/test/mountTest";

import Border from "..";

describe("Border working test", () => {
  vi.mock("~/pages/unAudit/hooks", () => {
    return {
      default: () => ({ unAuditUserQuery: { data: [] }, auditUser: () => {} }),
    };
  });
  vi.mock("~/pages/lab/hooks", () => {});
  vi.mock("~/pages/leave/hooks", () => {});

  mountTest(() => (
    <AppProvider>
      <Border />
    </AppProvider>
  ));

  it("should work", () => {
    render(
      <AppProvider>
        <Border />
      </AppProvider>
    );
    expect(
      screen.getByText("Ant Design ©2018 Created by Ant UED")
    ).toBeInTheDocument();
    expect(screen.getByText("待审核用户")).toMatchSnapshot();
  });
});
