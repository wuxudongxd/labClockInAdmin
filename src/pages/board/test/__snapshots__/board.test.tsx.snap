// Vitest Snapshot v1

exports[`Border working test > should work 1`] = `
<a
  href="/unaudit"
>
  待审核用户
</a>
`;
